File: SDKs/up.com.au/v1/java/lib/src/main/java/openapisdk/models/shared/MoneyObject.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * MoneyObject
 * Provides information about a value of money.
 * 
**/
public class MoneyObject {
    @JsonProperty("currencyCode")
    public String currencyCode;
    public MoneyObject withCurrencyCode(String currencyCode) {
        this.currencyCode = currencyCode;
        return this;
    }
    @JsonProperty("value")
    public String value;
    public MoneyObject withValue(String value) {
        this.value = value;
        return this;
    }
    @JsonProperty("valueInBaseUnits")
    public Long valueInBaseUnits;
    public MoneyObject withValueInBaseUnits(Long valueInBaseUnits) {
        this.valueInBaseUnits = valueInBaseUnits;
        return this;
    }
}

File: SDKs/up.com.au/v1/python/src/sdk/models/shared/moneyobject.py
import dataclasses
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclasses.dataclass
class MoneyObject:
    r"""MoneyObject
    Provides information about a value of money.
    
    """
    
    currency_code: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('currencyCode') }})
    value: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('value') }})
    value_in_base_units: int = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('valueInBaseUnits') }})
    
